sched/sched/sched_cpuselect.c

```diff
@@ -83,7 +83,7 @@ int nxsched_select_cpu(cpu_set_t affinity)
            * IDLE task is always the last task in the assigned task list.
            */
 
-          if (rtcb->flink == NULL)
+          if (is_idle_task(rtcb))
```

Comment:
As @masayuki2009 mentioned in failure report, not all the places need to use `this_cpu()` for idle task checking. For example here `pid` should be equal to `i` and not to `this_cpu()`. Let's examine other places more precisely.

In general I do not fully understand what is the issue that you are trying to fix with this PR. Is it a performance improvement of something else?

Reply:
@pkarashchenko

Reply:
@pkarashchenko This modify is not intended to fix bugs, but is just a preparation to support readytorun task list optimization.